src/command_line_parser.cc

```diff
@@ -1418,6 +1481,79 @@ TritonParser::Parse(int argc, char** argv)
+        std::cerr << "Warning: Overriding deprecated '--trace-rate' "
+                     "in favor of provided rate value in --trace-config!"
+                  << std::endl;
+      }
+      lparams.trace_rate_ = ParseOption<int>(global_setting.second);
+    }
+    if (global_setting.first == "level") {
+      if (trace_level_present) {
+        std::cerr << "Warning: Overriding deprecated '--trace-level' "
+                     "in favor of provided level in --trace-config!"
+                  << std::endl;
+      }
+      auto parsed_level_config = ParseTraceLevelOption(global_setting.second);
+      explicit_disable_trace |=
+          (parsed_level_config == TRITONSERVER_TRACE_LEVEL_DISABLED);
+      lparams.trace_level_ = static_cast<TRITONSERVER_InferenceTraceLevel>(
+          lparams.trace_level_ | parsed_level_config);
+    }
+    if (global_setting.first == "mode") {
+      lparams.trace_mode_ = ParseTraceModeOption(global_setting.second);
+    }
+    if (global_setting.first == "count") {
+      if (trace_count_present) {
+        std::cerr << "Warning: Overriding deprecated '--trace-count' "
+                     "in favor of provided count in --trace-config!"
+                  << std::endl;
+      }
+      lparams.trace_count_ = ParseOption<int>(global_setting.second);
+    }
+  }
+
+  if (lparams.trace_mode_ == TRITONSERVER_TRACE_MODE_OPENTELEMETRY) {
+    if (trace_filepath_present) {
+      std::cerr << "Warning: '--trace-file' is deprecated and will "
+                   "be ignored with opentelemetry tracing mode. "
+                << std::endl;
+    }
+    if (trace_log_frequency_present) {
+      std::cerr << "Warning: '--trace-log-frequency' is deprecated "
+                   "and will be ignored with opentelemetry tracing mode."
+                << std::endl;
+    }
+  } else if (lparams.trace_mode_ == TRITONSERVER_TRACE_MODE_TRITON) {
+    for (const auto& mode_setting : lparams.trace_config_map_[std::to_string(
+             TRITONSERVER_TRACE_MODE_TRITON)]) {
+      if (mode_setting.first == "file") {
+        if (trace_filepath_present) {
+          std::cerr << "Warning: Overriding deprecated '--trace-file' "
+                       "in favor of provided file in --trace-config!"
+                    << std::endl;
+        }
+        lparams.trace_filepath_ = mode_setting.second;
+      } else if (mode_setting.first == "log-frequency") {
+        if (trace_log_frequency_present) {
+          std::cerr << "Warning: Overriding deprecated '--trace-file' "
+                       "in favor of provided file in --trace-config!"
+                    << std::endl;
+        }
+        lparams.trace_log_frequency_ = ParseOption<int>(mode_setting.second);
+      }
+    }
+
+    if (lparams.trace_filepath_.empty()) {
+      throw ParseException(
+          "Error: file path must be provided for trace mode='triton' "
+          "Please set file path with --trace-config=triton,file=<filepath>");
+    }
```

Comment:
I think this logic needs to be checked, because the default trace mode is Triton, whether user passes CLI args or not. If they don't pass any trace CLI args (ex: tritonserver --model-repository models`), trace mode will be triton and this check will fail because no file was given.

Reply:
Agree, realized that later.